staging/src/k8s.io/client-go/examples/read-yaml-configuration-deployment/main.go

```diff
@@ -0,0 +1,214 @@
+	var deployment appsv1beta1.Deployment
+	err = json.Unmarshal(d, &deployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Read second resource - expecting service with size < 1024
+	yamlService := make([]byte, 1024)
+	_, err = yamlDecoder.Read(yamlService)
+	if err != nil {
+		panic(err)
+	}
+
+	// Trim unnecessary trailing 0x0 signs which are not accepted
+	TrimmedYamlService := strings.TrimRight(string(yamlService), string(byte(0)))
+
+	// Decode service resource from YAML to JSON
+	jsonService, _, err := decode([]byte(TrimmedYamlService), nil, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check "kind: service"
+	if jsonService.GetObjectKind().GroupVersionKind().Kind != "Service" {
+		panic(fmt.Sprintf("Error: Not \"Kind: Service\"\n%#v\n", jsonService))
+	}
+
+	// Marshall JSON Service
+	s, err := json.Marshal(&jsonService)
+	if err != nil {
+		panic(err)
+	}
+
+	//Unmarshall JSON into service struct
+	var service corev1.Service
+	err = json.Unmarshal(s, &service)
+	if err != nil {
+		panic(err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create clientset from outside of the cluster
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create client for deployments
+	deploymentsClient := clientset.AppsV1beta1().Deployments(corev1.NamespaceDefault)
+
+	// Create client for services
+	servicesClient := clientset.CoreV1().Services(corev1.NamespaceDefault)
+
+	// create deployment - try update on error
+	fmt.Printf("Create deployment %q\n", deployment.Name)
+	createdDeployment, err := deploymentsClient.Create(&deployment)
+	if err != nil {
+		fmt.Printf("Info: %s\n\n", err)
+
+		// update deployment
+		fmt.Printf("Update deployment %q\n", deployment.Name)
+		updatedDeployment, err := deploymentsClient.Update(&deployment)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Printf("Deployment %q updated\n\n", updatedDeployment.Name)
+	} else {
+		fmt.Printf("Deployment %q created\n\n", createdDeployment.Name)
+	}
+
+	// create service - try update on error
+	fmt.Printf("Create service %q\n", service.Name)
+	createdService, err := servicesClient.Create(&service)
+	if err != nil {
+		fmt.Printf("Info: %s\n\n", err)
+
+		// update service
+		fmt.Printf("Update service %q\n", service.Name)
+		updatedService, err := deploymentsClient.Update(&deployment)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Printf("Service %q updated\n", updatedService.Name)
+	} else {
+		fmt.Printf("Service %q created\n", createdService.Name)
+	}
+
+	// Get the port number by service's name
+	runningService, err := servicesClient.Get(service.Name, metav1.GetOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	// Get Pod "kube-addon-manager-minikube" of "kube-system" to retrieve 'minikube ip'
+	//
+	// Only valid for a minikube environment
+	//
+	pod, err := clientset.CoreV1().Pods("kube-system").Get("kube-addon-manager-minikube", metav1.GetOptions{})
```

Comment:
this is not portable. The example should work in every environment.

Reply:
No idea how to find out the IP address generally. Let's see what I can investigate tomorrow.